osier/models/dispatch.py

```diff
@@ -0,0 +1,210 @@
+import pandas as pd
+import numpy as np
+import pyomo.environ as pe
+import pyomo.opt as po
+from pyomo.environ import ConcreteModel
+from unyt import unyt_array
+import itertools as it
+
+
+class DispatchModel():
+    """
+    The :class:`DispatchModel` class creates and solves a basic dispatch
+    model from the perspective of a "grid operator." The model uses
+    `pyomo <https://pyomo.readthedocs.io/en/stable/index.html>`_
+    to create and solve a linear programming model.
+    The mathematical formulation for this problem is:
+
+    Minimize
+
+
+    .. math::
+        \\text{C}_{\\text{total}} = \\sum_t^T \\sum_u^U
+        \\left[c^{\\text{fuel}}_{u,t} + c^{\\text{om,var}}_{u,t}\\right]x_{u,t}
+
+    Such that,
+
+    1. The generation meets demand within a user-specified tolerance
+    (undersupply and oversupply)
+
+    .. math::
+        \\sum_u^Ux_{u,t} &\\geq \\left(1-\\text{undersupply}\\right)\\text{D}_t
+        \\ \\forall \\ t \\in T
+
+        \\sum_u^Ux_{u,t} &\\leq \\left(1+\\text{oversupply}\\right)\\text{D}_t 
+        \\ \\forall \\ t \\in T
+
+    2. A technology's generation (:math:`x_u`) does not exceed its capacity to
+    generate at any time, :math:`t`.
+
+    .. math::
+        x_{u,t} \\leq \\textbf{CAP}_{u} \\ \\forall \\ u,t \\in U,T
+
+
+    Parameters
+    ----------
+    technology_list : list of :class:`osier.Technology`
+        The list of :class:`Technology` objects to dispatch -- i.e. decide
+        how much energy each technology should produce.
+    net_demand : List, :class:`numpy.ndarray`, :class:`unyt.array.unyt_array`
+        The remaining energy demand to be fulfilled by the technologies in
+        :attr:`technology_list`. For example:
+
+        >>> from osier import DispatchModel
+        >>> from osier import Technology
+        >>> wind = Technology("wind")
+        >>> nuclear = Technology("nuclear")
+        >>> import numpy as np
+        >>> hours = np.arange(24)
+        >>> demand = np.cos(hours*np.pi/180)
+        >>> model = DispatchModel([wind, nuclear], demand)
+        >>> model.solve()
+
+    solver : str
+        Indicates which solver to use. May require separate installation.
+        Accepts: ['cplex']. Other solvers will be added in the future.
```

Comment:
You use `'cbc'` in `test_dispatch_model_solve`, so you should probably add that to your documentation.
```suggestion
        Accepts: ['cplex', 'cbc']. Other solvers will be added in the future.
```